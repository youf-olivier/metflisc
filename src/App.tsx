import Routes from 'routes';
import ThemeProvider from 'shared/components/ThemeProvider';

const App = () => {
  return (
    <ThemeProvider>
      <Routes />
    </ThemeProvider>
  );
};

export default App;
