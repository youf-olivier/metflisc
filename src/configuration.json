{
  "tmdbUrl": "https://api.themoviedb.org/3",
  "v3APIKey": "132e748598b28ef3f51cfbb17cbeb587",
  "V4BearerToken": "eyJhbGciOiJIUzI1NiJ9.eyJhdWQiOiIxMzJlNzQ4NTk4YjI4ZWYzZjUxY2ZiYjE3Y2JlYjU4NyIsInN1YiI6IjVmY2NjNjJjNzMxNGExMDA0MWNjYzJkYSIsInNjb3BlcyI6WyJhcGlfcmVhZCJdLCJ2ZXJzaW9uIjoxfQ.rC-dVYXV4uddUXS6N6ULF8oviE16oqf0ganH_coMv30",
  "api": {
    "popularEnpoint": "movie/popular",
    "searchEnpoint": "search/movie",
    "getEnpoint": "movie/"
  }
}
