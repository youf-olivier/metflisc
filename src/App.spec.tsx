import {
  render,
  screen,
  waitForElementToBeRemoved,
} from '@testing-library/react';
import UserEvent from '@testing-library/user-event';
import { setupServer } from 'msw/node';
import { rest } from 'msw';

import configuration from 'configuration.json';
import fixture from './Movies.fixture.json';
import App from './App';

const server = setupServer(
  rest.get(
    `${configuration.tmdbUrl}/${configuration.api.popularEnpoint}`,
    (req, res, ctx) => {
      return res(ctx.json(fixture.moviesList));
    },
  ),
);

beforeAll(() => server.listen({ onUnhandledRequest: 'error' }));
afterAll(() => server.close());
afterEach(() => server.resetHandlers());

describe('App tests suite', () => {
  it('Should render without crashing', async () => {
    const { asFragment } = render(<App />);
    await waitForElementToBeRemoved(() => screen.getByText(/pending/i));
    expect(asFragment()).toMatchSnapshot();
  });
  it('Should list all movies ', async () => {
    render(<App />);
    await waitForElementToBeRemoved(() => screen.getByText(/pending/i));
    expect(screen.getAllByRole('listitem')).toHaveLength(20);
  });
  it('Should go to detail when click on a movie', () => {});
  it('Should display filtered list when serach a movie', () => {});
  it('Should back to home when click on back button', async () => {
    render(<App />);
    const homeHeader = screen.getByRole('heading', { name: /Movie/i });
    expect(homeHeader).toBeInTheDocument();
    await waitForElementToBeRemoved(() => screen.getByText(/pending/i));
    UserEvent.click(screen.getAllByRole('listitem')[0]);

    const detailHeader = screen.getByRole('heading', { name: /DÃ©tail/i });
    expect(
      screen.queryByRole('heading', { name: /Movie/i }),
    ).not.toBeInTheDocument();
    expect(detailHeader).toBeInTheDocument();
  });
});
