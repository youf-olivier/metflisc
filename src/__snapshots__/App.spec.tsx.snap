// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App tests suite Should render without crashing 1`] = `
<DocumentFragment>
  .c3 {
  color: #ffffff;
  vertical-align: middle;
}

.c2 {
  display: inline-block;
  position: absolute;
  right: 0.5em;
  top: 50%;
  -webkit-transform: translate(0,-50%);
  -ms-transform: translate(0,-50%);
  transform: translate(0,-50%);
}

.c1 {
  color: #fff;
  font-size: 1.4em;
  font-weight: 500;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  text-align: center;
  display: inline-block;
}

.c0 {
  background-color: #60A5FA;
  text-align: center;
  position: relative;
}

@media (min-width:768px) {

}

@media (min-width:768px) {

}

@media (min-width:1024px) {

}

<div
    class="c0"
  >
    <h1
      class="c1"
    >
      Movies
    </h1>
    <div
      class="c2"
    >
      <i
        class="c3 material-icons"
      >
        wb_sunny
      </i>
      <span
        class="MuiSwitch-root material-icons MuiSwitch-sizeSmall"
      >
        <span
          aria-disabled="false"
          class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-1 MuiSwitch-switchBase PrivateSwitchBase-checked-2 Mui-checked"
        >
          <span
            class="MuiIconButton-label"
          >
            <input
              checked=""
              class="PrivateSwitchBase-input-4 MuiSwitch-input"
              type="checkbox"
              value=""
            />
            <span
              class="MuiSwitch-thumb"
            />
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </span>
        <span
          class="MuiSwitch-track"
        />
      </span>
      <i
        class="c3 material-icons"
      >
        brightness_3
      </i>
    </div>
  </div>
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  padding: 0.2em;
}

.c1 {
  display: contents;
}

.c2 {
  width: 48%;
  margin: 0.2em;
}

@media (min-width:768px) {
  .c0 {
    max-width: 950px;
    margin: auto;
  }
}

@media (min-width:768px) {
  .c2 {
    width: 24%;
  }
}

@media (min-width:1024px) {
  .c2 {
    width: 18%;
  }
}

<div
    class="c0"
    role="list"
  >
    <a
      aria-label="Jiu Jitsu"
      class="c1"
      href="/detail/590706"
      role="listitem"
    >
      <img
        alt="Jiu Jitsu"
        aria-label="Jiu Jitsu"
        class="c2"
        src="https://image.tmdb.org/t/p/w200///eLT8Cu357VOwBVTitkmlDEg32Fs.jpg"
      />
    </a>
    <a
      aria-label="Upside-Down Magic"
      class="c1"
      href="/detail/671583"
      role="listitem"
    >
      <img
        alt="Upside-Down Magic"
        aria-label="Upside-Down Magic"
        class="c2"
        src="https://image.tmdb.org/t/p/w200///xfYMQNApIIh8KhpNVtG1XRz0ZAp.jpg"
      />
    </a>
    <a
      aria-label="Tenet"
      class="c1"
      href="/detail/577922"
      role="listitem"
    >
      <img
        alt="Tenet"
        aria-label="Tenet"
        class="c2"
        src="https://image.tmdb.org/t/p/w200///k68nPLbIST6NP96JmTxmZijEvCA.jpg"
      />
    </a>
    <a
      aria-label="Fatman"
      class="c1"
      href="/detail/602211"
      role="listitem"
    >
      <img
        alt="Fatman"
        aria-label="Fatman"
        class="c2"
        src="https://image.tmdb.org/t/p/w200///4n8QNNdk4BOX9Dslfbz5Dy6j1HK.jpg"
      />
    </a>
    <a
      aria-label="The Dalton Gang"
      class="c1"
      href="/detail/729648"
      role="listitem"
    >
      <img
        alt="The Dalton Gang"
        aria-label="The Dalton Gang"
        class="c2"
        src="https://image.tmdb.org/t/p/w200///6OeGqp18oZucUGziMIRNhLouZ75.jpg"
      />
    </a>
    <a
      aria-label="Joker"
      class="c1"
      href="/detail/475557"
      role="listitem"
    >
      <img
        alt="Joker"
        aria-label="Joker"
        class="c2"
        src="https://image.tmdb.org/t/p/w200///udDclJoHjfjb8Ekgsd4FDteOkCU.jpg"
      />
    </a>
    <a
      aria-label="Magic Camp"
      class="c1"
      href="/detail/454433"
      role="listitem"
    >
      <img
        alt="Magic Camp"
        aria-label="Magic Camp"
        class="c2"
        src="https://image.tmdb.org/t/p/w200///awcPLFFYjufRXd2oAAP6ZIXF9vM.jpg"
      />
    </a>
    <a
      aria-label="Greenland"
      class="c1"
      href="/detail/524047"
      role="listitem"
    >
      <img
        alt="Greenland"
        aria-label="Greenland"
        class="c2"
        src="https://image.tmdb.org/t/p/w200///bNo2mcvSwIvnx8K6y1euAc1TLVq.jpg"
      />
    </a>
    <a
      aria-label="Mulan"
      class="c1"
      href="/detail/337401"
      role="listitem"
    >
      <img
        alt="Mulan"
        aria-label="Mulan"
        class="c2"
        src="https://image.tmdb.org/t/p/w200///aKx1ARwG55zZ0GpRvU2WrGrCG9o.jpg"
      />
    </a>
    <a
      aria-label="Hard Kill"
      class="c1"
      href="/detail/724989"
      role="listitem"
    >
      <img
        alt="Hard Kill"
        aria-label="Hard Kill"
        class="c2"
        src="https://image.tmdb.org/t/p/w200///ugZW8ocsrfgI95pnQ7wrmKDxIe.jpg"
      />
    </a>
    <a
      aria-label="The Christmas Chronicles: Part Two"
      class="c1"
      href="/detail/654028"
      role="listitem"
    >
      <img
        alt="The Christmas Chronicles: Part Two"
        aria-label="The Christmas Chronicles: Part Two"
        class="c2"
        src="https://image.tmdb.org/t/p/w200///6sG0kbEvAi3RRLcGGU5h8l3qAPa.jpg"
      />
    </a>
    <a
      aria-label="Chick Fight"
      class="c1"
      href="/detail/682377"
      role="listitem"
    >
      <img
        alt="Chick Fight"
        aria-label="Chick Fight"
        class="c2"
        src="https://image.tmdb.org/t/p/w200///4ZocdxnOO6q2UbdKye2wgofLFhB.jpg"
      />
    </a>
    <a
      aria-label="Peninsula"
      class="c1"
      href="/detail/581392"
      role="listitem"
    >
      <img
        alt="Peninsula"
        aria-label="Peninsula"
        class="c2"
        src="https://image.tmdb.org/t/p/w200///sy6DvAu72kjoseZEjocnm2ZZ09i.jpg"
      />
    </a>
    <a
      aria-label="Roald Dahl's The Witches"
      class="c1"
      href="/detail/531219"
      role="listitem"
    >
      <img
        alt="Roald Dahl's The Witches"
        aria-label="Roald Dahl's The Witches"
        class="c2"
        src="https://image.tmdb.org/t/p/w200///betExZlgK0l7CZ9CsCBVcwO1OjL.jpg"
      />
    </a>
    <a
      aria-label="Rogue City"
      class="c1"
      href="/detail/671039"
      role="listitem"
    >
      <img
        alt="Rogue City"
        aria-label="Rogue City"
        class="c2"
        src="https://image.tmdb.org/t/p/w200///9HT9982bzgN5on1sLRmc1GMn6ZC.jpg"
      />
    </a>
    <a
      aria-label="Rogue Warfare: Death of a Nation"
      class="c1"
      href="/detail/635780"
      role="listitem"
    >
      <img
        alt="Rogue Warfare: Death of a Nation"
        aria-label="Rogue Warfare: Death of a Nation"
        class="c2"
        src="https://image.tmdb.org/t/p/w200///8GVpIEBqlRBvx28G0LfEX0U8D2k.jpg"
      />
    </a>
    <a
      aria-label="Demon Slayer the Movie: Mugen Train"
      class="c1"
      href="/detail/635302"
      role="listitem"
    >
      <img
        alt="Demon Slayer the Movie: Mugen Train"
        aria-label="Demon Slayer the Movie: Mugen Train"
        class="c2"
        src="https://image.tmdb.org/t/p/w200///h8Rb9gBr48ODIwYUttZNYeMWeUU.jpg"
      />
    </a>
    <a
      aria-label="The SpongeBob Movie: Sponge on the Run"
      class="c1"
      href="/detail/400160"
      role="listitem"
    >
      <img
        alt="The SpongeBob Movie: Sponge on the Run"
        aria-label="The SpongeBob Movie: Sponge on the Run"
        class="c2"
        src="https://image.tmdb.org/t/p/w200///jlJ8nDhMhCYJuzOw3f52CP1W8MW.jpg"
      />
    </a>
    <a
      aria-label="Ava"
      class="c1"
      href="/detail/539885"
      role="listitem"
    >
      <img
        alt="Ava"
        aria-label="Ava"
        class="c2"
        src="https://image.tmdb.org/t/p/w200///qzA87Wf4jo1h8JMk9GilyIYvwsA.jpg"
      />
    </a>
    <a
      aria-label="Welcome to Sudden Death"
      class="c1"
      href="/detail/741067"
      role="listitem"
    >
      <img
        alt="Welcome to Sudden Death"
        aria-label="Welcome to Sudden Death"
        class="c2"
        src="https://image.tmdb.org/t/p/w200///elZ6JCzSEvFOq4gNjNeZsnRFsvj.jpg"
      />
    </a>
  </div>
</DocumentFragment>
`;
